SlowCheetah/Logging/ITransformationLogger.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Sayed Ibrahim Hashimi. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See  License.md file in the project root for full license information.
+
+namespace SlowCheetah
+{
+    using System;
+
+    /// <summary>
+    /// Interface for using an internal logger in <see cref="XmlTransformer"/>
```

Comment:
This will eventually be for `ITransformer` and not `XmlTransformer`. Keep that in mind so you remember to change it later.